lua/lualine/components/filetype.lua

```diff
@@ -8,7 +11,24 @@ FileType.update_status = function(self)
     local ok, devicons = pcall(require, 'nvim-web-devicons')
     if ok then
       local f_name, f_extension = vim.fn.expand('%:t'), vim.fn.expand('%:e')
-      self.options.icon = devicons.get_icon(f_name, f_extension)
+      icon, icon_highlight_group = devicons.get_icon(f_name, f_extension)
+      self.options.icon = icon
+
+      if self.options.icon and (self.options.colored or self.options.colored == nil) then
+        self.options.colored = true
+
+        local highlight_color = vim.fn.synIDattr(vim.fn.hlID(icon_highlight_group), 'fg')
+        local is_focused = vim.g.statusline_winid == vim.fn.win_getid() 
+        local default_highlight = highlight.format_highlight(is_focused, self.options.self.section)
+        local icon_highlight = self.options.self.section .. '_' .. icon_highlight_group
+        if not utils.highlight_exists(icon_highlight .. '_normal') then
```

Comment:
I made the assumption that checking if the highlight exists in normal mode is enough. Is this a safe assumption? @shadmansaleh

Reply:
> Assuming normal will work . But I think you're giving wrong arg to highlight exist as we prepend all highlight groups with 'lualine_'. Your query should have that too 

it already does, `icon_highlight` contains `self.options.self.section`

> Now I'm thinking since you can be sure once you call create_component_highlight the highlight will exist . A simpler solution might be to just keep track of what highlight groups you've created in self .

I don't think it would be simpler. This is one function call and we already track the created highlights there. Am I missing something here?

Reply:
> it already does, icon_highlight contains self.options.self.section

My bad I thought we were placeing only section names in self.section . Thanks for correcting me .

> I don't think it would be simpler. This is one function call and we already track the created highlights there. Am I missing something here?

ok this seems fine :)